ScatterPeas/scatterpeas/models.py

```diff
@@ -0,0 +1,632 @@
+        """Retrieves a reminder instance from reminder_id"""
+        if session is None:
+            session = DBSession
+        return session.query(Reminder).filter(Reminder.id == reminder_id).one()
+
+    @classmethod
+    def get_next_job(cls, reminder_id, session=None):
+        """Takes a reminder_id to retrieve current rrule set and generate
+        the next job in datetime format for utc. Returns None if there is
+        no next job."""
+        if session is None:
+            session = DBSession
+        reminder = cls.retrieve_instance(reminder_id)
+        dtstart = RRule.get_rrules(reminder.rrule.id)
+        rrule_gen = rrule(0, dtstart=dtstart, count=1)
+        now = datetime.utcnow()
+        next_job = rrule_gen.after(now)
+        if next_job is None:
+            return None
+        else:
+            return next_job
+
+    @classmethod
+    def create_reminder(cls, alias_id=0, title="",
+                        text_payload="", media_payload="",
+                        rstate=True, session=None):
+        '''Instantiates a Reminder instance. An RRule object is required
+        for the Reminder. We currently require any parent function to
+        attach an rrule_id to the Reminder after instantiation; the
+        Reminder object is not valid without this
+        '''
+        if session is None:
+            session = DBSession
+        if title != "" and alias_id != 0:
+            instance = cls(title=title, alias_id=alias_id,
+                           text_payload=text_payload,
+                           media_payload=media_payload, rstate=rstate)
+            session.add(instance)
+            session.flush()
+            return instance
+
+
+class RRule(Base):
+    """Contains iCal specification rules corresponding to an event. This
+    class is not yet fully implemented, but stands in for future
+    expandability"""
+    __tablename__ = 'rrules'
+
+    # constrain this to primary key of reminders
+    id = Column(Integer, primary_key=True, nullable=False, autoincrement=True)
+    reminder_id = Column(Integer, ForeignKey('reminders.id'))
+    dtstart = Column(DateTime)
+
+    @classmethod
+    def get_rrules(cls, rrule_id, session=None):
+        if session is None:
+            session = DBSession
+        # Passes back just dtstart for now; future implementations should
+        # probably use a dictionary
+        return session.query(RRule).filter(RRule.id == rrule_id).one().dtstart
+
+    @classmethod
+    def create_rrule(cls, reminder_id=0,
+                     dtstart=(datetime.utcnow() + timedelta(hours=24)),
+                     session=None):
+        if session is None:
+            session = DBSession
+        try:
+            # This try block can disappear when we refactor to use the ORM
+            # more idiomatically
+            Reminder.retrieve_instance(reminder_id)
+        except NoResultFound:
+            raise NoResultFound
```

Comment:
Seems like you could just let the database throw an integrity error when you try to create an rrule with a reminder that doesn't exist.